### DISCUSSION TITLE
Javascript

### DISCUSSION THREAD

**Discussion by MK-Hasan-Mahdi** [MK-Hasan-Mahdi/jspractice#1] [Q&A] [answered]
How Javascript works?

**Comment by ImRanKhan81m** [comment, marked as answer]
The way JavaScript works is interesting. Inside a normal Web page you place some JavaScript code (See How Web Pages Work for details on Web pages). When the browser loads the page, the browser has a built-in interpreter that reads the JavaScript code it finds in the page and runs it.

**Comment by AyonJD** [comment]
The source code is passed through a program called a compiler, which translates it into bytecode that the machine understands and can execute. In contrast, JavaScript has no compilation step. Instead, an interpreter in the browser reads over the JavaScript code, interprets each line, and runs it.
